source/developers/transforms-with-object-lambda.rst

```diff
@@ -0,0 +1,353 @@
+
+.. code-block:: shell
+   :class: copyable
+
+   MINIO_LAMBDA_WEBHOOK_ENABLE_myfunction=on MINIO_LAMBDA_WEBHOOK_ENDPOINT_myfunction=http://localhost:5000 minio server /data
+
+Replace ``myfunction`` with the name of your handler function and ``/data`` with the location of the MinIO directory for your local deployment. 
+The output resembles the following:
+
+.. code-block:: shell
+
+   MinIO Object Storage Server
+   Copyright: 2015-2023 MinIO, Inc.
+   License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
+   Version: RELEASE.2023-03-24T21-41-23Z (go1.19.7 linux/arm64)
+   
+   Status:         1 Online, 0 Offline. 
+   API: http://192.168.64.21:9000  http://127.0.0.1:9000       
+   RootUser: minioadmin 
+   RootPass: minioadmin 
+   Object Lambda ARNs: arn:minio:s3-object-lambda::myfunction:webhook 
+
+
+Test the Handler
+~~~~~~~~~~~~~~~~
+
+To test the Lambda handler function, first create an object to transform.
+Then invoke the handler, in this case with ``curl``, using the presigned URL from a Go function.
+
+#. Create a bucket and object for the handler to transform.
+
+   .. code-block:: shell
+      :class: copyable
+
+      mc alias set myminio/ http://localhost:9000 minioadmin minioadmin
+      mc mb myminio/myfunctionbucket
+      cat > testobject << EOF
+      Hello, World!
+      EOF
+      mc cp testobject myminio/myfunctionbucket/
+
+#. Invoke the Handler
+
+   The following Go code uses the :doc:`The MinIO Go SDK </developers/go/minio-go>` to generate a presigned URL and print it to ``stdout``.
+
+   .. code-block:: go
+      :class: copyable
+
+      package main
+
+      import (
+         "context"
+         "log"
+         "net/url"
+         "time"
+         "fmt"
+
+         "github.com/minio/minio-go/v7"
+         "github.com/minio/minio-go/v7/pkg/credentials"
+      )
+
+      func main() {
+
+         // Connect to the MinIO deployment
+         s3Client, err := minio.New("localhost:9000", &minio.Options{
+            Creds:  credentials.NewStaticV4("my_admin_user", "my_admin_password", ""),
+            Secure: false,
+         })
+         if err != nil {
+            log.Fatalln(err)
+         }
+
+         // Set the Lambda function target using its ARN
+         reqParams := make(url.Values)
+         reqParams.Set("lambdaArn", "arn:minio:s3-object-lambda::myfunction:webhook")
+
+         // Generate a presigned url to access the original object
+         presignedURL, err := s3Client.PresignedGetObject(context.Background(), "myfunctionbucket", "testobject", time.Duration(1000)*time.Second, reqParams)
+         if err != nil {
+            log.Fatalln(err)
+         }
+	 
+         // Print the URL to stdout
+         fmt.Println(presignedURL)
+      }      
+
+   In the code above, replace the following values:
+
+   - Replace ``my_admin_user`` and ``my_admin_password`` with root credentials for a MinIO deployment
```

Comment:
Can you use a non-root credential to do transforms?

Reply:
Hmm. No policy fails, as expected. And `readwrite` succeeds. But `readonly` is also ok?
```
$ mc admin policy attach myminio readonly --user=feorlen
Policy `readonly` successfully attached to user `feorlen`
$ curl -v $(go run presigned.go)
[...]
hELLO, wORLD!
```

@harshavardhana what permissions should/should not work for triggering Object Lambda?

Reply:
yes you are only doing "readonly" there is no WRITE anywhere in the code in lambda layer.